src/Connection.ts

```diff
@@ -15,28 +15,43 @@ export interface MomoConnectionOptions {
 }
 
 export class Connection {
-  private executionsRepository?: ExecutionsRepository;
+  private schedulesRepository?: SchedulesRepository;
   private jobRepository?: JobRepository;
 
-  constructor(private readonly mongoClient: MongoClient, private readonly collectionsPrefix?: string) {}
+  constructor(
+    private readonly mongoClient: MongoClient,
+    private readonly pingIntervalMs: number,
+    private readonly scheduleId: string,
+    private readonly collectionsPrefix?: string
+  ) {}
 
-  static async create({ url, collectionsPrefix }: MomoConnectionOptions): Promise<Connection> {
-    const connection = new Connection(new MongoClient(url), collectionsPrefix);
+  static async create(
+    { url, collectionsPrefix }: MomoConnectionOptions,
+    pingIntervalMs: number,
+    scheduleId: string
+  ): Promise<Connection> {
+    const connection = new Connection(new MongoClient(url), pingIntervalMs, scheduleId, collectionsPrefix);
 
-    await connection.connect();
+    await connection.mongoClient.connect();
+
+    const schedulesRepository = connection.getSchedulesRepository();
+    await schedulesRepository.createIndex();
+    const jobRepository = connection.getJobRepository();
+    await jobRepository.createIndex();
 
     return connection;
   }
 
-  getExecutionsRepository(deadExecutionsThreshold = 60 * 1000): ExecutionsRepository {
-    if (!this.executionsRepository) {
-      this.executionsRepository = new ExecutionsRepository(
+  getSchedulesRepository(): SchedulesRepository {
+    if (!this.schedulesRepository) {
```

Comment:
Is the `schedulesRepository` ever `undefined` if we've called `create` beforehand? 

And do we actually ever want this class to exist without having `create` called on it right after intantiation?

Reply:
Ditto for the `getJobRepository`.

Reply:
Thanks, I refactored it and now it's much nicer :D